File: app/src/main/java/com/example/livestreamingagora/viewmodel/LoginViewModel.java
package com.example.livestreamingagora.viewmodel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;

import com.example.livestreamingagora.ConstantsKt;
import com.example.livestreamingagora.repository.LoginRepository;
import com.example.livestreamingagora.models.LoginBody;

public class LoginViewModel extends AndroidViewModel {
    private LoginRepository repository;
    public LoginViewModel(@NonNull Application application) {
        super(application);
        repository = new LoginRepository(application);
    }

    public void userLogin(LoginBody body, LoginRepository.LoginCallBack callBack){
        repository.userLogin(callBack, body);
    }

    public void userLogout( LoginRepository.LoginCallBack callBack){
        repository.userLogout(callBack);
    }

    public void userDeActiveUser(LoginRepository.LoginCallBack callBack,String channelName){
        repository.deActiveUser(callBack,channelName);
    }
}

File: app/src/main/java/com/example/livestreamingagora/viewmodel/LoginViewModel.kt
package com.example.livestreamingagora.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.example.livestreamingagora.models.LoginBody
import com.example.livestreamingagora.repository.LoginRepository
import com.example.livestreamingagora.repository.LoginRepository.LoginCallBack

class LoginViewModel(application: Application) : AndroidViewModel(application) {
    private val repository: LoginRepository

    init {
        repository = LoginRepository(application)
    }

    fun userLogin(body: LoginBody?, callBack: LoginCallBack?) {
        callBack?.let {
            repository.userLogin(it, body) }
    }

    fun userLogout(callBack: LoginCallBack?) {
        if (callBack != null) {
            repository.userLogout(callBack)
        }
    }

    fun userDeActiveUser(callBack: LoginCallBack?, channelName: String?) {
        callBack?.let {
            repository.deActiveUser(it, channelName) }
    }
}
